packages/block-library/src/navigation/edit/use-convert-classic-menu-to-block-menu.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * WordPress dependencies
+ */
+import { useRegistry } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { useState, useCallback } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useCreateNavigationMenu from './use-create-navigation-menu';
+import menuItemsToBlocks from '../menu-items-to-blocks';
+
+const CLASSIC_MENU_CONVERSION_SUCCESS = 'success';
+const CLASSIC_MENU_CONVERSION_ERROR = 'error';
+const CLASSIC_MENU_CONVERSION_PENDING = 'pending';
+
+function useConvertClassicToBlockMenu( clientId ) {
+	const createNavigationMenu = useCreateNavigationMenu( clientId );
+	const registry = useRegistry();
+
+	const [ status, setStatus ] = useState( 'idle' );
+	const [ value, setValue ] = useState( null );
+	const [ error, setError ] = useState( null );
+
+	async function convertClassicMenuToBlockMenu( menuId, menuName ) {
+		let navigationMenu;
+		let classicMenuItems;
+
+		const menuItemsParameters = {
```

Comment:
This is used only once – let's inline it.

Reply:
`menuItemsParameters` is now inlined.